lib/Frontend/ArgsToFrontendOptionsConverter.cpp

```diff
@@ -310,6 +310,40 @@ bool ArgsToFrontendOptionsConverter::convert(
                      "-enable-library-evolution");
   }
 
+  Opts.AllowNonResilientAccess = Args.hasArg(OPT_experimental_allow_non_resilient_access);
+  if (Opts.AllowNonResilientAccess) {
+    // Override the option to skip non-exportable decls.
+    if (Opts.SkipNonExportableDecls) {
+      Diags.diagnose(SourceLoc(), diag::warn_ignore_option_overriden_by,
+                     "-experimental-skip-non-exportable-decls",
+                     "-experimental-allow-non-resilient-access");
+      Opts.SkipNonExportableDecls = false;
+    }
+    // Overriden by enable-testing as it already bypasses resilience.
+    if (Opts.EnableTesting) {
+      Diags.diagnose(SourceLoc(), diag::warn_ignore_option_overriden_by,
```

Comment:
I'm confused by this because testability and non-resilient access seem compatible with each other. Imagine you are the developer specifying `-experimental-allow-non-resilient-access` in your project settings. If you have a configuration in which your library is built for testability to facilitate unit testing, this diagnostic would be telling you that in that configuration you ought to be removing `-experimental-allow-non-resilient-access` from the command line, which makes the project configuration awkward. Why do we need to set `AllowNonResilientAccess = false` here?

Reply:
They are compatible but the intention here was to make it known that package specific optimization isn't needed if testing or private import is enabled, but maybe this is more misleading than helpful. Will remove these.